File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/angajatPersistence/RepoBilet.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using log4net;
using proiectSCcsharp.repository.interfaces;
using proiectSCcsharp.domain;

namespace proiectSCcsharp.repository
{
    public class RepoBilet : IRepoBilet
    {
        private static readonly ILog logger = LogManager.GetLogger("RepoBilet");
        IDictionary<String, string> props;

        public RepoBilet(IDictionary<string, string> props)
        {
            logger.Info("Initializing RepoBilet class");
            this.props = props;
        }

        public Bilet add(Bilet entity)
        {
            logger.InfoFormat("Saving Bilet...", entity);

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "INSERT INTO Bilete(id_bilet, cumparator_name, id_spectacol, nr_seats) VALUES (@i, @c, @ids, @n)";

                var paramId = command.CreateParameter();
                paramId.ParameterName = "@i";
                paramId.Value = null;
                command.Parameters.Add(paramId);

                var nameParameter = command.CreateParameter();
                nameParameter.ParameterName = "@c";
                nameParameter.Value = entity.cumparatorName;
                command.Parameters.Add(nameParameter);

                var idsParameter = command.CreateParameter();
                idsParameter.ParameterName = "@ids";
                idsParameter.Value = entity.idSpectacol;
                command.Parameters.Add(idsParameter);

                var seatsParameter = command.CreateParameter();
                seatsParameter.ParameterName = "@n";
                seatsParameter.Value = entity.nrSeats;
                command.Parameters.Add(seatsParameter);

                var result = command.ExecuteNonQuery();

                return entity;
            }
        }
        public Bilet remove(long aLong)
        {
            logger.InfoFormat("Deleting Bilet...", aLong);

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "DELETE FROM Bilete WHERE id_bilet=@i";

                var idParameter = command.CreateParameter();
                idParameter.ParameterName = "@i";
                idParameter.Value = aLong;
                command.Parameters.Add(idParameter);

                var result = command.ExecuteNonQuery();

                return findElement(aLong);
            }
        }
        public List<Bilet> getAll()
        {
            logger.InfoFormat("Finding all Bilete...");

            List<Bilet> bilete = new List<Bilet>();

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM Bilete";

                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        long id = dataReader.GetInt64(0);
                        string name = dataReader.GetString(1);
                        long ids = dataReader.GetInt64(2);
                        int seats = dataReader.GetInt32(3);
                        Bilet bilet = new Bilet(id, name, ids, seats);
                        bilete.Add(bilet);
                    }
                }
            }

            return bilete;
        }

        public Bilet findElement(long id)
        {
            logger.InfoFormat("Entering findElement with value {0}", id);

            Bilet bilet = null;

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM Bilete WHERE id_bilet = @i";

                var idParameter = command.CreateParameter();
                idParameter.ParameterName = "@i";
                idParameter.Value = id;
                command.Parameters.Add(idParameter);

                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        long id2 = dataReader.GetInt64(0);
                        string name = dataReader.GetString(1);
                        long ids = dataReader.GetInt64(2);
                        int seats = dataReader.GetInt32(3);
                        bilet = new Bilet(id2, name, ids, seats);
                        logger.InfoFormat("Found bilet: {0}", bilet);
                        return bilet;
                    }
                }
            }
            return null;
        }
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 7-servicii REST/Java/Rest/proiect/Persistence/src/main/java/proiect/persistence/repository/jdbc/RepoBilet.java
package proiect.persistence.repository.jdbc;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import proiect.domain.Bilet;
import proiect.persistence.repository.IRepoBilet;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoBilet implements IRepoBilet {
    private static final Logger logger= LogManager.getLogger();
    private final JDBCUtils jdbcUtils;

    public RepoBilet(Properties properties) {
        logger.info("Initializing RepoBilet with properties: {} ",properties);
        jdbcUtils=new JDBCUtils(properties);
    }

    @Override
    public Bilet add(Bilet entity) {
        logger.traceEntry("Saving bilet {}", entity);
        String query = "INSERT INTO Bilete(cumparator_name,id_spectacol,nr_seats) VALUES(?,?,?)";
        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query)
        ) {
            statement.setString(1, entity.getCumparatorName());
            statement.setInt(2, entity.getIdSpectacol());
            statement.setInt(3, entity.getNrSeats());

            int result=statement.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException e) {
            logger.error(e);
            throw new RuntimeException(e);
        }
        logger.traceExit();
        return entity;
    }
    @Override
    public Bilet remove(Integer aLong) {
        logger.traceEntry("Deleting bilet {}", aLong);
        String query = "DELETE FROM Bilete WHERE id_bilet = ?";

        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query)
        ) {
            statement.setInt(1, aLong);
            int result=statement.executeUpdate();
            logger.trace("Deleted {} instance", result);
        } catch (SQLException e) {
            logger.error(e);
            throw new RuntimeException(e);
        }
        logger.traceExit();
        return findElement(aLong);
    }

    @Override
    public Bilet findElement(Integer aLong) {
        logger.traceEntry();
        String query = "SELECT * FROM Bilete WHERE id_bilet=?";
        Bilet bilet = null;
        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query)
        ) {
            statement.setInt(1, aLong);
            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                int id=resultSet.getInt("id_bilet");
                String name = resultSet.getString("cumparator_name");
                int id_spectacol = resultSet.getInt("id_spectacol");
                int nr_seats = resultSet.getInt("nr_seats");
                bilet=new Bilet(id,name,id_spectacol,nr_seats);
            }
        } catch (SQLException e) {
            logger.error(e);
            e.printStackTrace();
        }
        logger.traceExit();
        return bilet;
    }

    @Override
    public Iterable<Bilet> getAll() {
        logger.traceEntry();
        List<Bilet> bilete = new ArrayList<>();

        String query = "SELECT * from Bilete";
        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query);
             ResultSet resultSet = statement.executeQuery()
        ) {
            while (resultSet.next()) {
                int id=resultSet.getInt("id_bilet");
                String name = resultSet.getString("cumparator_name");
                int id_spectacol = resultSet.getInt("id_spectacol");
                int nr_seats = resultSet.getInt("nr_seats");

                Bilet bilet=new Bilet(id,name,id_spectacol,nr_seats);
                bilete.add(bilet);
            }
        } catch (SQLException e) {
            logger.error(e);
            e.printStackTrace();
        }
        logger.traceExit();
        return bilete;
    }
}
